add support for scaleway service discovery (#8555)
prometheus/prometheus: docs/configuration/configuration.md

for a practical example on how to set up your Eureka app and your Prometheus
configuration.

### `<scaleway_sd_config>`

```yaml

> server: <string>

Reviewer: You are missing the zone, tags, etc fields here.

for a practical example on how to set up your Eureka app and your Prometheus
configuration.

### `<scaleway_sd_config>`

The following meta labels are available on targets during [relabeling](#relabel_config):

#### Instance role

* `__meta_scaleway_instance_id`: the id of the instance
* `__meta_scaleway_instance_ip_is_public`: whether the ip of the target is publicly routed
* `__meta_scaleway_instance_ip_version`: version of the ip `IPv4`, `IPv6`
* `__meta_scaleway_instance_image_name`: name of the server image
* `__meta_scaleway_instance_name`: name of the server
* `__meta_scaleway_instance_status`: status of the server
* `__meta_scaleway_instance_tags`: tag list of the server
* `__meta_scaleway_instance_type`: commercial type of the server

> * `__meta_scaleway_instance_zone`: zone of the instance (`fr-par-1`, `fr-par-2`, `nl-ams-1`, `pl-waw-1`)

Reviewer: I would give an exemple but not list them all, to avoid going out of date.

Author: Is there automated reference documentation that would register all possible meta? If there is one I can add a link to it to have the exhaustive list. Otherwise, I think it is best to have all the labels and enforce to keep them up to date.

* `__meta_scaleway_baremetal_ip_version`: version of the ip `IPv4`, `IPv6`
* `__meta_scaleway_baremetal_name`: name of the server
* `__meta_scaleway_baremetal_os_name`: name of the os used
* `__meta_scaleway_baremetal_os_version`: version of the os used
* `__meta_scaleway_baremetal_status`: status of the server
* `__meta_scaleway_baremetal_tags`: tag list of the server
* `__meta_scaleway_baremetal_type`: commercial type of the server
* `__meta_scaleway_baremetal_zone`: zone of the server (`fr-par-1`, `fr-par-2`, `nl-ams-1`, `pl-waw-1`)

See below for the configuration options for Scaleway discovery:

```yaml
# The port to scrape metrics from.
[ port: <int> | default = 80 ]

# Refresh interval to re-read the targets list.
[ refresh_interval: <duration> | default = 60s ]

# API URL to use when doing the List requests
[ api_url: <string> | default = "https://api.scaleway.com" ]

# Secret key to use when listing servers. https://console.scaleway.com/project/credentials
[ secret_key: <secret> ]

# Project id to use when listing servers. https://console.scaleway.com/project/credentials
[ project_id: <string> ]

# Role can be either instance or baremetal

> [ role: <string> ]

Reviewer: ```suggestion
role: <string>
```

This is mandatory, should not be surrounded by `[]`

> # Zone specify the availability zone of the server you target (fr-par-1, fr-par-2, nl-ams-1, pl-waw-1, ...)

Reviewer: I would add an exemple and then a link to the API doc where you have the up to date list iof the zones.

> [ zone: <string> ]

Reviewer: ```suggestion
[ zone: <string> | default = fr-par-1 ]
```

for a practical example on how to set up your Eureka app and your Prometheus
configuration.

### `<scaleway_sd_config>`

The following meta labels are available on targets during [relabeling](#relabel_config):

#### Instance role

* `__meta_scaleway_instance_id`: the id of the instance
* `__meta_scaleway_instance_ip_is_public`: whether the ip of the target is publicly routed
* `__meta_scaleway_instance_ip_version`: version of the ip `IPv4`, `IPv6`
* `__meta_scaleway_instance_image_name`: name of the server image
* `__meta_scaleway_instance_name`: name of the server
* `__meta_scaleway_instance_status`: status of the server

> * `__meta_scaleway_instance_tags`: tag list of the server

Reviewer: ```suggestion
* `__meta_scaleway_instance_tags`: the list of tags of the target joined by the tag separator
```

* `__meta_scaleway_instance_type`: commercial type of the server

> * `__meta_scaleway_instance_zone`: zone of the instance (ex: `fr-par-1`, complete list on <https://developers.scaleway.com/en/>)

Reviewer: ```suggestion
* `__meta_scaleway_instance_zone`: zone of the instance (ex: `fr-par-1`, complete list at [https://developers.scaleway.com/en/](https://developers.scaleway.com/en/))
```

#### Baremetal role

* `__meta_scaleway_baremetal_id`: the id of the instance
* `__meta_scaleway_baremetal_ip_is_public`: whether the ip of the target is publicly routed
* `__meta_scaleway_baremetal_ip_version`: version of the ip `IPv4`, `IPv6`
* `__meta_scaleway_baremetal_name`: name of the server
* `__meta_scaleway_baremetal_os_name`: name of the os used
* `__meta_scaleway_baremetal_os_version`: version of the os used
* `__meta_scaleway_baremetal_status`: status of the server
* `__meta_scaleway_baremetal_tags`: tag list of the server
* `__meta_scaleway_baremetal_type`: commercial type of the server

> * `__meta_scaleway_baremetal_zone`: zone of the server (ex: `fr-par-1`, complete list on <https://developers.scaleway.com/en/>)

Reviewer: ```suggestion
* `__meta_scaleway_baremetal_zone`: zone of the server (ex: `fr-par-1`, complete list at [https://developers.scaleway.com/en/](https://developers.scaleway.com/en/))
```

See below for the configuration options for Scaleway discovery:

```yaml
# The port to scrape metrics from.
[ port: <int> | default = 80 ]

# Refresh interval to re-read the targets list.
[ refresh_interval: <duration> | default = 60s ]

# API URL to use when doing the List requests
[ api_url: <string> | default = "https://api.scaleway.com" ]

> # Secret key to use when listing servers. https://console.scaleway.com/project/credentials

Reviewer: ```suggestion
# Secret key to use when listing targets. https://console.scaleway.com/project/credentials
```

[ secret_key: <secret> ]

> # Project id to use when listing servers. https://console.scaleway.com/project/credentials

Reviewer: ```suggestion
# Project ID of the targets.
```

[ project_id: <string> ]

> # Role can be either instance or baremetal

Reviewer: ```suggestion
# Role of the targets to retrieve. Must be `instance` or `baremetal`.
```

role: <string>

> # Zone specify the availability zone of the server you target (fr-par-1, fr-par-2, nl-ams-1, pl-waw-1, ...)

Reviewer: ```suggestion
# Zone is the availability zone of your targets (e.g. fr-par-1).
```

[ zone: <string> | default = fr-par-1 ]

> # Name specify a name filter to apply on the ListServers request

Reviewer: ```suggestion
# Name specify a name filter to apply on the ListServers request.
```

That is blurry to me, and will probably be the same for users. ListServer is maybe too low level? maybe add an example?

> [ name: <string> ]

Reviewer: Should we call that name_filter? I am mostly afraid that users believe this is the name of the service discovery.

> # Tags specify a tag filter to apply on the ListServers requests

Reviewer: Commends must end with full stop.

Can you specify if hosts should have ALL the tags or ANY of them?

tags:

> [ - <string> ... ]

Reviewer: ```suggestion
[ - <string> ]
```

openstack_sd_configs:
  [ - <openstack_sd_config> ... ]

# List of Scaleway service discovery configurations.
scaleway_sd_configs:

>   [ - <scaleway_sd_config> ... ]

Author: I think it is added https://github.com/prometheus/prometheus/pull/8555/files#diff-17f1012e0c2fbd9bcd8dff3c23b18ff4b6676eef3beca6f8a3e72e6a36633334R247
